File: tests/fixtures/dart-strategy-pos/lib/pricing.dart
// Strategy pattern applied: interface + per-variant classes + map dispatch.
abstract interface class PricingStrategy {
  double price(double base);
}

class Regular implements PricingStrategy {
  @override double price(double base) => base;
}

class Vip implements PricingStrategy {
  @override double price(double base) => base * 0.8;
}

class Staff implements PricingStrategy {
  @override double price(double base) => base * 0.5;
}

final _strategies = <String, PricingStrategy>{
  'regular': Regular(), 'vip': Vip(), 'staff': Staff(),
};

double price(String kind, double base) => _strategies[kind]!.price(base);

double discountedTotal(String kind, double base, int qty) =>
    price(kind, base) * qty;

File: tests/fixtures/swift-strategy-pos/Sources/App/Pricing.swift
// Strategy pattern applied: protocol + per-variant structs + dictionary dispatch.
public protocol PricingStrategy { func price(base: Double) -> Double }

public struct Regular: PricingStrategy { public func price(base: Double) -> Double { base } }
public struct Vip:     PricingStrategy { public func price(base: Double) -> Double { base * 0.8 } }
public struct Staff:   PricingStrategy { public func price(base: Double) -> Double { base * 0.5 } }

private let strategies: [String: any PricingStrategy] = [
    "regular": Regular(), "vip": Vip(), "staff": Staff(),
]

public func price(kind: String, base: Double) -> Double {
    guard let s = strategies[kind] else { fatalError("unknown kind: \(kind)") }
    return s.price(base: base)
}

public func discountedTotal(kind: String, base: Double, qty: Int) -> Double {
    price(kind: kind, base: base) * Double(qty)
}
